src/bin/jcli_app/utils/io.rs

```diff
@@ -1,37 +1,48 @@
+use std::fs::File;
+use std::io::{stdin, stdout, BufRead, BufReader, Error as IoError, Write};
 use std::path::Path;
 
-/// open the given file path as a writable stream, or stdout if no path
+/// open the given file path as a readable stream, or stdin if no path
 /// provided
-pub fn open_file_write<P: AsRef<Path>>(path: &Option<P>) -> Box<dyn std::io::Write> {
-    if let Some(path) = path {
-        Box::new(
-            std::fs::OpenOptions::new()
-                .create(true)
-                .write(true)
-                .read(false)
-                .append(false)
-                .open(path)
-                .unwrap(),
-        )
-    } else {
-        Box::new(std::io::stdout())
+pub fn open_file_read<P: AsRef<Path>>(path: &Option<P>) -> Result<impl BufRead, Error> {
+    match path {
+        Some(path) => {
+            let reader = File::open(path).map_err(|e| Error::new_open_file_read(path, e))?;
+            Ok(Box::new(BufReader::new(reader)) as Box<dyn BufRead>)
+        }
+        None => Ok(Box::new(BufReader::new(stdin())) as Box<dyn BufRead>),
     }
 }
 
-/// open the given file path as a readable stream, or stdin if no path
+/// open the given file path as a writable stream, or stdout if no path
 /// provided
-pub fn open_file_read<P: AsRef<Path>>(path: &Option<P>) -> Box<dyn std::io::BufRead> {
-    if let Some(path) = path {
-        Box::new(std::io::BufReader::new(
-            std::fs::OpenOptions::new()
-                .create(false)
-                .write(false)
-                .read(true)
-                .append(false)
-                .open(path)
-                .unwrap(),
-        ))
-    } else {
-        Box::new(std::io::BufReader::new(std::io::stdin()))
```

Comment:
This Diff seems like you changed the order of the functions... this is not necessary